### PR TITLE
Dex whitelist

### PR DESCRIPTION
**Pull request by fbielejec**
# Description

- Add a query
- whitelist all possible pairs exchange contract deployment

### CODE REVIEW

**Review by h4nsu — APPROVED** [review]

File: contracts/simple_dex/lib.rs [hunk 1]
```diff
@@ -322,6 +322,11 @@ mod simple_dex {
             Ok(())
         }
 
+        #[ink(message)]
+        pub fn can_swap_pair(&self, from: AccountId, to: AccountId) -> bool {
```
Comment (deuszx) [inline contracts/simple_dex/lib.rs:326]: Please add documentation on the public endpoint. It gets compiled into `metadata.json` file and may be useful for frontend wallets like Contracts UI.

File: e2e-tests/src/test/button_game/mod.rs [hunk 2]
```diff
@@ -159,6 +160,12 @@ pub fn simple_dex(config: &Config) -> Result<()> {
 
     dex.swap(account_conn, token1, balance_after, token3, mega(90))?;
     assert_recv_id(&mut events, "Swapped");
+
+    // can't swap a pair not on the whitelist
+
+    dex.remove_swap_pair(authority_conn, token3.into(), token1.into())?;
+    assert_recv_id(&mut events, "SwapPairRemoved");
```
Comment (deuszx) [inline e2e-tests/src/test/button_game/mod.rs:167]: Why had the test below been failing before you added these two lines? i.e. it [the test] asserts that there was no trade made and I'm assuming that it used to pass so why did it?

Comment (fbielejec) [reply e2e-tests/src/test/button_game/mod.rs:167]: cause before pair was not on a whitelist for swapping

File: .github/scripts/run_smartnet.sh [hunk 3]
```diff
@@ -2,7 +2,8 @@
 
 set -e
 
-export NODE_IMAGE=public.ecr.aws/p6e8q1z1/aleph-node:latest
+export NODE_IMAGE=573243519133.dkr.ecr.us-east-1.amazonaws.com/feature-env-aleph-node:fe-benjamin_c643069
```
[the remark this answers is not in the document]
Comment (fbielejec) [reply .github/scripts/run_smartnet.sh:5]: i don't know, maybe @mikogs does

File: contracts/simple_dex/lib.rs (hunk 1 above)
Comment (fbielejec) [reply contracts/simple_dex/lib.rs:327]: done

**Comment by obrok** [comment]
Duh, I already forgot that we have the separate marketplace thing _that I wrote myself_ :/

**Pull request merged**